model/messages/consensus.go

```diff
@@ -1,16 +1,30 @@
 package messages
 
 import (
+	"github.com/onflow/flow-go/model"
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// BlockProposal is part of the consensus protocol and represents the the leader
+// BlockProposal is part of the consensus protocol and represents the leader
 // of a consensus round pushing a new proposal to the network.
 type BlockProposal struct {
 	Header  *flow.Header
 	Payload *flow.Payload
 }
 
+// StructureValid checks basic structural validity of the message and ensures no required fields are nil.
+func (m *BlockProposal) StructureValid() error {
+	if m.Header == nil {
+		return model.NewStructureInvalidError("nil block header")
+	}
+	if m.Payload == nil {
```

Comment:
Payload has lists of pointers, it's possible that adversary constructs a list of nil pointers, probably that will get parsed correctly but most likely will crash the node with segmentation fault

Reply:
Added in https://github.com/onflow/flow-go/pull/3470/commits/8b47a64b24ac42d4f3df2b7c2a4ed1f6cacec715